src/cart/crt_init.c

```diff
@@ -442,6 +442,22 @@ crt_str_to_provider(const char *str_provider)
 			provider_idx = crt_na_dict[i].nad_type;
 			break;
 		}
+		if (is_primary && crt_na_dict[i].nad_type == CRT_PROV_UCX &&
+		    !strncmp(str_provider, CRT_UCX_STR, strlen(CRT_UCX_STR))) {
+			if (strlen(str_provider) > strlen(CRT_UCX_STR) && strchr(str_provider, '+')) {
+				D_ALLOC(p, strlen(str_provider)+1);
+				if (!p) {
+					D_WARN("Unable to set UCX provider to %s, using default %s",
```

Comment:
suggestion not to add extra warnings on alloc failures - otherwise you will hit new warnings under fault injection testing when they purposely fail allocs and track messages printed

Reply:
Sounds good to me. It isn't even a problem necessarily if the allocation fails (which won't happen). We just use the existing default.